libraries/conf.rb

```diff
@@ -98,7 +98,7 @@ def whyrun_supported?
         execute "#{r.name}.#{r.conf_type}-systemd-reload" do
           command 'systemctl daemon-reload'
           action :nothing
-          only_if { r.is_a?(Chef::Resource::SystemdUnit) }
+          only_if { r.is_a?(Chef::Resource::SystemdUnit) && r.auto_reload }
           subscribes :run, "file[#{conf_path}]", :immediately
```

Comment:
I think one of the issues with that is if you have future parts of your cookbook that depends on the changes in the unit.